pkg/sidecar/sidecar.go

```diff
@@ -38,103 +36,69 @@ type Config struct {
 	RenewSignal              string `hcl:"renewSignal"`
 	Timeout                  string `hcl:"timeout"`
 	ReloadExternalProcess    func() error
+	Log                      logger.Logger
 }
 
 // Sidecar is the component that consumes the Workload API and renews certs
 // implements the interface Sidecar
 type Sidecar struct {
-	config            *Config
-	processRunning    int32
-	process           *os.Process
-	workloadAPIClient workload.X509Client
-	certReadyChan     chan struct{}
+	config         *Config
+	processRunning int32
+	process        *os.Process
+	certReadyChan  chan struct{}
+	ErrChan        chan error
 }
 
 const (
-	// default timeout Duration for the workloadAPI client when the defaultTimeout
-	// is not configured in the .conf file
-	defaultTimeout = 5 * time.Second
-	delayMin       = time.Second
-	delayMax       = time.Minute
-
 	certsFileMode = os.FileMode(0644)
 	keyFileMode   = os.FileMode(0600)
 )
 
 // NewSidecar creates a new SPIFFE sidecar
-func NewSidecar(config *Config) (*Sidecar, error) {
-	timeout, err := getTimeout(config)
-	if err != nil {
-		return nil, err
+func NewSidecar(config *Config) *Sidecar {
+	if config.Log == nil {
+		config.Log = logger.Null
 	}
-
 	return &Sidecar{
-		config:            config,
-		workloadAPIClient: newWorkloadAPIClient(config.AgentAddress, timeout),
-		certReadyChan:     make(chan struct{}, 1),
-	}, nil
+		config:        config,
+		certReadyChan: make(chan struct{}),
+		ErrChan:       make(chan error, 1),
+	}
 }
 
 // RunDaemon starts the main loop
 // Starts the workload API client to listen for new SVID updates
 // When a new SVID is received on the updateChan, the SVID certificates
 // are stored in disk and a restart signal is sent to the proxy's process
 func (s *Sidecar) RunDaemon(ctx context.Context) error {
-	// Create channel for interrupt signal
-	errorChan := make(chan error, 1)
-
-	updateChan := s.workloadAPIClient.UpdateChan()
-
-	// start the workloadAPIClient
+	client, err := workloadapi.New(ctx, workloadapi.WithAddr("unix://"+s.config.AgentAddress),
```

Comment:
nit: If we're just using this client once, to start up a watcher, it would be better to use the top-level [WatchX509Context](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2@v2.1.1/workloadapi#WatchX509Context)  convenience function.